homeassistant/components/switch/ecoplug.py

```diff
@@ -0,0 +1,85 @@
+"""
+EcoPlugs Component for aut-discovering Wion and EcoPlugs automatically.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/ecoplug/
+"""
+
+import logging
+
+from datetime import timedelta
+
+from homeassistant.const import (DEVICE_DEFAULT_NAME, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.components.switch import SwitchDevice
+
+DEFAULT_INVERT_LOGIC = False
+
+DOMAIN = 'ecoplug'
+REQUIREMENTS = ['pyecoplug==0.0.5']
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=5)
+
+
+class EcoPlugSwitch(SwitchDevice):
+    """Return the polling state."""
+
+    def __init__(self, plug):
+        """Initialize the switch."""
+        self._plug = plug
+        self._name = plug.name or DEVICE_DEFAULT_NAME
+        self._state = self._plug.is_on()
```

Comment:
Init state as `None`and instead call `add_devices` like so:
```py
add_devices(devices, True)
```
That will update the entity during entity addition.